src/luacov/reporter.lua

```diff
@@ -4,24 +4,40 @@
 -- @name luacov.reporter
 local reporter = {}
 
+--- Raw version of string.gsub
+local function replace(s, old, new)
+   old = old:gsub("%p", "%%%0")
+   new = new:gsub("%%", "%%%%")
+   return (s:gsub(old, new))
+end
+
+local fixups = {
+   { " ", " +" }, -- ' ' represents "at least one space"
+   { "=", " *= *" }, -- '=' may be surrounded by spaces
+   { "(", " *%( *" }, -- '(' may be surrounded by spaces
+   { ")", " *%) *" }, -- ')' may be surrounded by spaces
+   { "<ID>", " *[%w_]+ *" }, -- identifier
+   { "<FULLID>", " *[%w_][%w_%.%[%]0-9]+ *" }, -- identifier, possibly indexed
+   { "<BEGIN_LONG_STRING>", "%[(=*)%[[^]]* *" },
+   { "<IDS>", "[%w_, ]+" }, -- comma-separated identifiers
+   { "<ARGS>", "[%w_, \"'%.]*" }, -- comma-separated arguments
+   { "<FIELDNAME>", "%[? *[\"'%w_]+ *%]?" }, -- field, possibly like ["this"]
+   { " * ", " " }, -- collapse consecutive spacing rules
+   { " + *", " +" }, -- collapse consecutive spacing rules
+   { " ", "%s" }, -- accept all whitespace characters as space
+}
+
 --- Utility function to make patterns more readable
 local function fixup(pat)
-   return pat:gsub(" ", " +")                  -- ' ' represents "at least one space"
-             :gsub("=", " *= *")               -- '=' may be surrounded by spaces
-             :gsub("%(", " *%%( *")            -- '(' may be surrounded by spaces
-             :gsub("%)", " *%%) *")            -- ')' may be surrounded by spaces
-             :gsub("<ID>", " *[%%w_]+ *")      -- identifier
-             :gsub("<FULLID>", " *[%%w_][%%w_.%%[%%]0-9]+ *") -- identifier
-             :gsub("<BEGIN_LONG_STRING>", "%%[(=*)%%[[^]]* *")
-             :gsub("<IDS>", "[%%w_, ]+")       -- comma-separated identifiers
-             :gsub("<ARGS>", "[%%w_, \"'%%.]*") -- comma-separated arguments
-             :gsub("<FIELDNAME>", "%%[? *[\"'%%w_]+ *%%]?") -- field, possibly like ["this"]
-             :gsub(" %* ", " ")                -- collapse consecutive spacing rules
-             :gsub(" %+ %*", " +")             -- collapse consecutive spacing rules
-end
-
-local long_string_1 = "^() *" .. fixup"<FULLID>=<BEGIN_LONG_STRING>$"
-local long_string_2 = "^() *" .. fixup"local <FULLID>=<BEGIN_LONG_STRING>$"
+   for _, fixup_pair in ipairs(fixups) do
+      pat = replace(pat, fixup_pair[1], fixup_pair[2])
+   end
+
+   return pat
+end
+
+local long_string_1 = "^()" .. fixup " *<FULLID>=<BEGIN_LONG_STRING>$"
```

Comment:
Why the move of `" *"` into the fixup argument string? Won't this expand to `" +*"` ?